reactioncommerce/reaction-next-starterkit
# chore: update typography variants based on MUI warning

Impact: **minor**
Type: **chore**

## Issue
The following warning is appearing inside starterkit. MUI is updating their typography styles, and we were using older styles.
```
Warning: Material-UI: you are using the deprecated typography variants that will be removed in the next major release.
Please read the migration guide under https://material-ui.com/style/typography#migration-to-typography-v2
```

## Solution
This PR updates all variants to use their new naming conventions, based on the link provided: https://material-ui.com/style/typography#migration-to-typography-v2

The only changes that were noticible at first glance were the Product Title on the PDP. I've added some customization there to keep it looking the same. Everywhere else, simply replacing the variant name kept it the same style.

## Breaking changes
Any customized plugins / code will need to be updated to accommodate new typography variants

## Testing
1. Start up the app, and see that it looks the same as it did on any other branch/
1. See typography warning is gone from console.
1. I compared the pages side by side using https://storefront.staging.reactioncommerce.com/
